### DISCUSSION TITLE
ArgoCD cannot access repo using proxy

### DISCUSSION THREAD

**Discussion by oyuskeseliev** [argoproj/argo-cd#6921] [Q&A] [answered]
Hi all,

I`m trying to configure ArgoCD deployment to use the corporate proxy for accessing Azure DevOPs repo using SSH keys.

I've created the following configmap:

```
apiVersion: v1
data:
  HTTP_PROXY: 10.123.66.254:8080
  HTTPS_PROXY: 10.123.66.254:8080
  NO_PROXY: argocd-dex-server,argocd-metrics,argocd-redis,argocd-redis-ha-announce-0,argocd-redis-ha-announce-1,argocd-redis-ha-announce-2,argocd-redis-ha-haproxy,argocd-repo-server,argocd-server,argocd-server-metrics,cluster.local,10.0.0.0/8
  http_proxy: 10.XXX.XXX.XXX:8080
  https_proxy: 10.XXX.XXX.XXX:8080
  no_proxy: argocd-dex-server,argocd-metrics,argocd-redis,argocd-redis-ha-announce-0,argocd-redis-ha-announce-1,argocd-redis-ha-announce-2,argocd-redis-ha-haproxy,argocd-repo-server,argocd-server,argocd-server-metrics,cluster.local,10.0.0.0/8
kind: ConfigMap
metadata:
  creationTimestamp: "2021-08-06T12:34:10Z"
  name: proxy-config
  namespace: argocd
  resourceVersion: "1573726"
  uid: f2ee0c4f-0f8f-4718-8d82-25eb657ea0cd
```

and applied the config map as follows:
```
kubectl set env -n argocd deployment/argocd-server --from=configmap/proxy-config --containers='*'
kubectl set env -n argocd deployment/argocd-repo-server --from=configmap/proxy-config --containers='*'
```

for some reason ArgoCD is still not respecting the values:
<img width="380" alt="image" src="https://user-images.githubusercontent.com/15718906/128514688-ee2fa916-de6f-41d0-8fa6-80b7544b3e59.png">

any help will be greatly appreciated, as I cannot find a clear guidelines for proxy configuration.

Regards,
Ognyan

**Comment by jannfis** [comment, marked as answer]
Hi @oyuskeseliev. Accessing Git repositories via SSH using a proxy is not currently supported, and SSH over HTTP proxy probably will never be supported either.

You should have more luck when you are able to connect the Git repository using HTTPS instead.

**Reply by oyuskeseliev** [reply]
@jannfis Thanks for the clarification. I will go with HTTPS.
